src/stream/src/executor/aggregation/agg_state.rs

```diff
@@ -112,16 +98,19 @@ impl<S: StateStore> AggState<S> {
         ops: Ops<'_>,
         visibility: Option<&Bitmap>,
         columns: &[&ArrayImpl],
-        agg_state_table: Option<&mut AggStateTable<S>>,
+        storage: &mut AggStateStorage<S>,
     ) -> StreamExecutorResult<()> {
         debug_assert!(verify_chunk(ops, visibility, columns));
         match self {
-            Self::Value(state) => state.apply_chunk(ops, visibility, columns),
+            Self::Value(state) => {
+                debug_assert!(matches!(storage, AggStateStorage::ResultValue));
+                state.apply_chunk(ops, visibility, columns)
+            }
             Self::MaterializedInput(state) => {
-                let agg_state_table =
-                    agg_state_table.expect("State table is expected for materialized input state");
+                let state_table =
+                    must_match!(storage, AggStateStorage::MaterializedInput { table, .. } => table);
```

Comment:
Will `derive(enum_as_inner::EnumAsInner)` help?

Reply:
Seems `enum_as_inner` is not that convenient for enum variant with named fields. With that we have to write `let (table, _) = storage.as_materialized_input().unwrap()`, which I think is not very explicit. And `must_match` can be less intrusive and more flexible. Anyway, both are OK to me.